indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -572,19 +585,22 @@ public Response getCompleteTasks(
         )
     );
 
-    final List<TaskStatusPlus> completeTasks = recentlyFinishedTasks
-        .stream()
-        .map(status -> new TaskStatusPlus(
-            status.getId(),
-            taskFunction.apply(status.getId()).getType(),
-            taskStorageQueryAdapter.getCreatedTime(status.getId()),
-            // Would be nice to include the real queue insertion time, but the TaskStorage API doesn't yet allow it.
-            DateTimes.EPOCH,
-            status.getStatusCode(),
-            status.getDuration(),
-            TaskLocation.unknown())
-        )
-        .collect(Collectors.toList());
+    final List<TaskStatusPlus> completeTasks = Lists.newArrayList(Iterables.transform(
```

Comment:
Suggest keeping with the java 8 fluency style here. Does it need to be removed for some reason?

Reply:
It was removed as we wanted to make call to the storage layer only once.
taskStorageQueryAdapter.getCreatedDateAndDataSource(status.getId());